### ISSUE TITLE
Missing django_session table

### ISSUE THREAD

**Issue by jooola** [LibreTime/libretime#1370]
**Describe the bug**

While trying to lo in the django api browser, I get a HTTP 500 because the django_session table is missing.

```
2021-09-30 15:46:33,939 log ERROR Internal Server Error: /api-auth/login/
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UndefinedTable: relation "django_session" does not exist
LINE 1: SELECT (1) AS "a" FROM "django_session" WHERE "django_sessio...
                               ^


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/views/generic/base.py", line 70, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/utils/decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/views/decorators/debug.py", line 89, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/utils/decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/utils/decorators.py", line 130, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/utils/decorators.py", line 43, in _wrapper
    return bound_method(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/contrib/auth/views.py", line 63, in dispatch
    return super().dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/views/generic/base.py", line 98, in dispatch
    return handler(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/views/generic/edit.py", line 142, in post
    return self.form_valid(form)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/contrib/auth/views.py", line 92, in form_valid
    auth_login(self.request, form.get_user())
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/contrib/auth/__init__.py", line 111, in login
    request.session.cycle_key()
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/contrib/sessions/backends/base.py", line 344, in cycle_key
    self.create()
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/contrib/sessions/backends/db.py", line 51, in create
    self._session_key = self._get_new_session_key()
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/contrib/sessions/backends/base.py", line 196, in _get_new_session_key
    if not self.exists(session_key):
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/contrib/sessions/backends/db.py", line 47, in exists
    return self.model.objects.filter(session_key=session_key).exists()
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/models/query.py", line 808, in exists
    return self.query.has_results(using=self.db)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/models/sql/query.py", line 552, in has_results
    return compiler.has_results()
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/models/sql/compiler.py", line 1145, in has_results
    return bool(self.execute_sql(SINGLE))
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/models/sql/compiler.py", line 1175, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.7/dist-packages/Django-3.2.7-py3.7.egg/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: relation "django_session" does not exist
LINE 1: SELECT (1) AS "a" FROM "django_session" WHERE "django_sessio...

```

**To Reproduce**

1. Log in the django api browser.

**LibreTime version:** main branch the 2021-09-30

**Comment by paddatrapper** [comment]
Sounds like the migrations didn't run. Are there any django related tables in the DB?

**Comment by Rakshitjn** [comment]
> Sounds like the migrations didn't run. Are there any django related tables in the DB?

I think you are right, you are using the table which do not exist

**Comment by jooola** [comment]
> Sounds like the migrations didn't run. Are there any django related tables in the DB?

None of the tables exists.

**Comment by jooola** [comment]
I ran the migrations manually and the tables are now present, so I suppose this was intended to be automatic, so we might need to leave this open for further investigation.

**Comment by paddatrapper** [comment]
The migrations happen as part of the install script IIRC

**Comment by jooola** [comment]
@paddatrapper I don't think so, let me search, but I saw it part of the database setup in PHP.

**Comment by jooola** [comment]
Here:
https://github.com/LibreTime/libretime/blob/1db244114cb39351cc80f493835519c3834b2e71/airtime_mvc/public/setup/database-setup.php

**Comment by jooola** [comment]
I just ran a full fresh install on buster, everything is working as expected, the tables are present. So I am unsure of this is from a previous error in my install process.

So I guess we need to move this to the installation script, as I think it safe to consider `libretime-api migrate` as idempotent.

**Comment by paddatrapper** [comment]
Ah, yes. It is run in the Web installer section after the database has been created. Potentially the issue here is on upgrade from a instance from before django to current where the migrations are not run. I don't think we handle that case in the installer...

**Comment by paddatrapper** [comment]
We can't move it to the installer, as the DB hasn't been created, so we can't call the migrations. However, we need to run the migrations on upgrade in the installer script

**Comment by jooola** [comment]
Some time ago I started to work on some libretime control utility, for backups, basic project managements, and possibly a setup script.

The idea is to move the php setup to the CLI setup script, and move some parts of the install script in the setup script.

I think it is ugly to create a `airtime` user that can create any db. We should rather create a single DB X with a single user Y that has access to it.

This will also allow to configure elements in a single place, and not spread it between the install script and the Php setup form.

We can also setup the global config file on the fly after we create/configure the DB for example, or RabbitMQ.

TLDR; We should properly split `Install` and `Setup/Configure` scripts.

**Comment by jooola** [comment]
I haven't faced this anymore, if it happens again I will reopen, but I feel this might have been the result of a bad setup on my side.

**Comment by paddatrapper** [comment]
I'll open a new issue around handling migrations on upgrade

**Comment by jooola** [comment]
Do you mean similar to this https://github.com/LibreTime/libretime/issues/1377 ?

**Comment by paddatrapper** [comment]
Not the propel migrations, but the Django ones
